### PR TITLE
Treat any non 200 status as failure for thingiverse

### LINKED ISSUE #900: Dead thingiverse images are not filtered out because they return 403
## Description

When we filter dead links, we do not mark image as dead when the direct URL returns 403 or 429 because that might be caused by the provider rate limiting. However, for Thingiverse, a 403 should probably be a reason for marking the link as dead. Otherwise, we get a lot of dead links on the frontend.

## Reproduction

1. Search for cat images.
2. Select the `Thingiverse` source filter.
3. You should see an "Image not available" thumbnail as the first result. There are more than 5 such thumbnails in the first 3 pages. If you click on "Get this image", you get to a 404 page. And the direct URL returns 403.
[API result for such an image](https://api.openverse.engineering/v1/images/236f4e8d-92a7-44ac-9283-0c5bce26f23d/)

## Screenshots

Screenshot 2023-03-13 at 9 34 09 AM

## Additional context

There might be other providers who return 403 for dead links.

I marked this as low as I only know of one provider that this issue affects, and it's usually not very prominent in the results.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #900 by @obulat

## Description

Fixes the linked issue with the strategy discussed in the issue.

I also went ahead and renamed `validate_images` to `check_dead_links`. I chose a dead-link-specific name because the method is already quite complex, and I wanted to avoid the assumption that additional validations could get loaded into that process. If that's a point of contention though, I can change it to just `validate_results` or something.

## Testing Instructions

Run the app locally and ensure it runs. There are no thingiverse results locally so you'll need to rely on the unit tests to confirm that specific behaviour.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/catalog/api/utils/check_dead_links.py [hunk 1]
```diff
@@ -119,7 +119,14 @@ def validate_images(
     for idx, _ in enumerate(cached_statuses):
         del_idx = len(cached_statuses) - idx - 1
         status = cached_statuses[del_idx]
-        if status == 429 or status == 403:
+        # thingiverse treated as failure despite the suspect status code
```
Comment (obulat) [inline api/catalog/api/utils/check_dead_links.py:122]: We could extract a function like `is_request_rate_limited_or_blocked(status, provider)` to make this part of code easier to read - but this is not a blocking comment.

**Review by obulat — APPROVED** [review]
I tested this with this data from Thingiverse: https://gist.github.com/obulat/a5e927202ef187dc89caaed301490dd7, and the dead links were filtered out.

I didn't know that `filter_dead` is not `true` by default, so it took me some time to understand why the images are not filtered out at first.

**Comment by sarayourfriend** [comment]
@krysal @dhruvkb are either of y'all available to review this PR for the high priority bug?

**Review by dhruvkb — APPROVED** [review]
LGTM, appreciate the renaming of the function to better represent what it does.

**Pull request merged**